src/main/java/com/mackervoy/calum/mud/TDBStore.java

```diff
@@ -23,4 +29,70 @@ public final static boolean inUseLocation(String fileLocation) {
 		File f = new File(fileLocation);
 		return TDBStore.inUseLocation(f);
 	}
+	
+	/**
+	 * takes an unsafe URI String and returns the path part of the URI if valid, or null if not
+	 */
+	private final static Optional<String> getFilePathFromUri(String uriInput) {
+		try {
+			URL uri = new URL(uriInput);
+			
+			if(!TDBStore.isLocalURI(uriInput)) {
+				throw new NotFoundException("The given dataset is not hosted on this site!");
+			}
+			
+			return Optional.of(uri.getPath().substring(1));
+		}
+		catch(MalformedURLException e) {
+			return Optional.empty();
+		}
+	}
+	
+	private final static String getFullFilePath(String filePath) {
+		String root = MUDApplication.getRootDirectory();
+		
+		// normalise style
+		if(filePath.startsWith("./")) filePath = filePath.substring(2);
+		if(root.startsWith("./")) root = root.substring(2);
+		
+		if(filePath.startsWith(root)) return filePath;
+		return root + filePath;
+	}
+	
+	/**
+	 * @param uriOrFilePath a URI or filesystem path for the dataset item
+	 * @return the DatasetItem found at this resource
+	 * @throws NotFoundException if it can't be found
+	 * example URI: http://localhost:8080/mud/world/#something
+	 * example file paths: ./mud/world/, /mud/world/ or world/
+	 */
+	public final static DatasetItem getDatasetItem(String uriOrFilePath) {
+		
+		// convert a URI to a file system path
+		String filePath = getFilePathFromUri(uriOrFilePath).orElse(uriOrFilePath);
+		
+		if(!TDBStore.inUseLocation(new File(getFullFilePath(filePath)))) {
+			throw new NotFoundException("The given dataset is not in use!");
+		}
+		
+		String[] splitString = filePath.split("/");
```

Comment:
This sorta thing feels like we should be delegating it to a third party library (or wrapping it), to keep our codebase focused on the business logic in the app - e.g. https://commons.apache.org/proper/commons-io/javadocs/api-1.4/org/apache/commons/io/FilenameUtils.html#getName(java.lang.String)

Reply:
Yeah spot on :+1:

Reply:
Updated this MR to at least use the `FilenameUtils` in string manipulation and such in `TDBStore` and `AbstractFileStorageWrapper`... does this cover it?

In terms of the general feature.. I'm not sure how much of the datastore management the platform should be responsible for and how much we can outsource: #32 

In terms of the purpose of the repo I think this repository is responsible for providing a Java library to build a MUD application, but that a particular server with custom features would use + extend it .. I'm not sure where the line is - I opened this about it if you have thoughts #34